### ISSUE TITLE
Add client-side and serverless support for Tailwind

### ISSUE THREAD

**Issue by Titou325** [OnedocLabs/react-print-pdf#7]
As of now, Tailwind can only be used with `react-print-pdf` in a Node environment as Tailwind relies on the file structure. This has caused [issues](https://github.com/OnedocLabs/react-print-pdf/issues/5) when bundled alongside the other components.

We want to port Tailwind to client-side.

**Comment by Titou325** [comment]
As mentioned in #5 and #11, enabling client-side support for Tailwind will iron out issues in most serverless environments as well.

Right now we are collecting the Tailwind css classes by parsing the React tree but the `postcss` plugin requires a file system to generate the classes. Existing ports of the plugin to client-side (https://github.com/mhsdesign/browser-jit-generate-tailwindcss) are asynchronous, which won't work in the current React rendering pipe.

We will post updates on this issue as we dig deeper into the Tailwind toolkit.

**Comment by Titou325** [comment]
Serverless support is fixed, client side remains flaky.

**Comment by Titou325** [comment]
@the-homeless-god @eliasamaral I have also added a Vite preview at https://github.com/Titou325/react-print-test-browser to show the current progress regarding client-side HTML generation including Tailwind support.

**Comment by Titou325** [comment]
Hey @judygab!

The `compile` function in `@fileforge/react-print` will compile a React component to a portable HTML string (that is, including styles and eventual assets). If you don't need the PDF, you may not need a portable bundle. In this case, you are not required to use the `compile` function. If you need a PDF, you will need to use either the [Fileforge API](https://docs.fileforge.com/sdk-reference/node/pdf/generate-from-html) or another HTML to PDF provider to convert. We will be providing more snippets very soon!

If you are creating a live preview in the browser, you may be better off without using the `compile` function altogether, especially if you have Tailwind already set-up.

In short:

* Use `compile` if you are generating a PDF (that will require a further step), or require a portable HTML,
* Don't use `compile` (or not needed) if you want to render components part of a larger app, which already has a styling pipeline in place.

Happy to chat on Discord if needed!
